src/Core/Cart/Calculator.php

```diff
@@ -150,10 +150,11 @@ public function getTotal($ignoreProcessedFlag = false)
 
         $amount = new AmountImmutable();
         foreach ($this->cartRows as $cartRow) {
-            $rowPrice = $cartRow->getFinalTotalPrice();
+            $rowPrice = $cartRow->getInitialTotalPrice();
             $amount = $amount->add($rowPrice);
         }
-        $shippingFees = $this->fees->getFinalShippingFees();
+        $amount = $amount->sub($this->getDiscountTotal());
+        $shippingFees = $this->fees->getInitialShippingFees();
```

Comment:
So if you add `getInitialShippingFees` when is the shipping discount applied? Also I don't know what wrapping fees are but should it be considered when computing the max authorized discount?

Reply:
mmm, my bad it should already be inside the total discount! forget this comment